src/app/workflow/launch-third-party/multi-cloud-launch/multi-cloud-launch.component.ts

```diff
@@ -141,32 +142,52 @@ export class MultiCloudLaunchComponent extends Base implements OnInit {
         );
       }
 
-      const newCustomInstance: CloudInstance = {
-        url: this.launchWith.url,
-        partner: this.languagePartner,
-        supportsFileImports: null,
-        supportsHttpImports: null,
-        supportedLanguages: new Array<Language>(),
-        displayName: url.hostname,
-      };
-
-      this.usersService.postUserCloudInstance(this.user.id, newCustomInstance).subscribe(
-        (usersCloudInstances: Array<CloudInstance>) => {
-          this.usersCloudInstances = usersCloudInstances;
-        },
-        (error: HttpErrorResponse) => {
-          // It's okay for the save to the db to fail for now. At this step, we only care about what URL's users
-          // are trying to use. Probably most of these failures will be the for violating the uniqueness constraint
-          console.log(error.message);
-        }
-      );
+      if (url) {
+        const newCustomInstance: CloudInstance = {
+          url: this.launchWith.url,
+          partner: this.languagePartner,
+          supportsFileImports: null,
+          supportsHttpImports: null,
+          supportedLanguages: new Array<Language>(),
+          displayName: url.hostname,
+        };
+
+        this.usersService.postUserCloudInstance(this.user.id, newCustomInstance).subscribe(
+          (usersCloudInstances: Array<CloudInstance>) => {
+            this.usersCloudInstances = usersCloudInstances;
+          },
+          (error: HttpErrorResponse) => {
+            // It's okay for the save to the db to fail for now. At this step, we only care about what URL's users
+            // are trying to use. Probably most of these failures will be the for violating the uniqueness constraint
+            console.log(error.message);
+          }
+        );
+      } else {
+        // Remove link to prevent navigating to invalid URL
+        document.getElementById('multiCloudLaunchButton').removeAttribute('href');
```

Comment:
I didn't try running your branch and don't 100% follow the code flow, but does this ever need to get added back? What if the user subsequently specifies a good/valid url, if that makes sense?

Reply:
I don't think we need to manually add it back. `this.launchWith.url` is binded to `href` [here](https://github.com/dockstore/dockstore-ui2/blob/08b5b9f7ddf4398a728b6b454ea0390c96bc37b8/src/app/workflow/launch-third-party/multi-cloud-launch/multi-cloud-launch.component.html#L62) so if a user specifies a different URL (valid or invalid) then the `href` property is updated to the new URL